src/Directive.php

```diff
@@ -29,10 +32,27 @@ abstract class Directive
     public const WAKE_LOCK = 'wake-lock';
     public const XR = 'vr';
 
-    public static function isValid(string $directive): bool
+    public static function make(string $directive, string $type = 'default'): DirectiveContract
+    {
+        return match ($type) {
+            'default' => DefaultDirective::directive($directive),
+            default => throw InvalidDirective::unknownDirectiveType($type),
+        };
+    }
+
+    public function addRule(string $rule): static
     {
-        $constants = (new ReflectionClass(static::class))->getConstants();
```

Comment:
No longer using reflection to check if a directive is valid, will be handled by the `make(..)` method.
Personally I'm not a fan of using reflection, adds quite a bit of magic. Which could be just as easy be a regular class or similar.

I feel that when we're not using reflection, that it will benefit the clarity of the code as a whole.